### PR TITLE
[CAS-1066] Message list component doc

### PR DESCRIPTION
**Pull request by ogkuzmin**
### 🎯 Goal

Docs for our `MessageListView`

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (CMS, cookbooks, tutorial)
- [x] Reviewers added

### CODE REVIEW

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:27]: I cannot comment above this line but I would replace this sentence ```MessageListView comes with a set of actions out of the box that are available by long-pressing a message. There are multiple available actions:``` with ```MessageListView comes with a set of actions available out of the box by long-pressing a message:```. Moreover some of the listed actions are hidden by default and we also allow to Unmute users

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:125]:
```suggestion
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:126]:
```suggestion
:::note
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:130]:
```suggestion
:::
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:80]:
```suggestion
Default action handlers are set after binding ViewModel with the View. You can customize default behavior by overriding each of the following handlers:
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:149]: I think we can point to Dokka instead

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:151]: This is probably no longer needed

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [reply docusaurus/docs/Android/03-ui/04-components/03-message-list.md:80]: and maybe before that - what do you think about explaining what handlers are?

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:134]: It might be worth it writing that the same as with Handlers - default ones are set when binding view model with view

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:159]:
```suggestion
`MessageListView` provides a quite big set of xml attributes available for customization. The full list of them is available [here](https://github.com/GetStream/stream-chat-android/blob/main/stream-chat-android-ui-components/src/main/res/values/attrs_message_list_view.xml).
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:206]: We could use:
:::note
:::

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:234]: NIT: I would use unordered list

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:239]: Do we have other flags? Maybe in that case it's worth it to enumerate all of them?

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:256]:
```suggestion
`MessageListItem` provides API for creating custom view holders. All you need to do is:
1. Extend  `MessageListItemViewHolderFactory`
2. Write your own logic for creating ViewHolders 
3. Make a new factory instance and set it to `MessageListView`.
```

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (adasiewiczr) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:257]:
```suggestion
Let's consider an example when we want to create custom view holders for messages that came from other users less than 24 hours ago.

```

**Review by adasiewiczr — COMMENTED** [review]
Really good job! Docs look great overall, screenshots are amazing!
 I've just added small comments and suggestions

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (ogkuzmin) [reply docusaurus/docs/Android/03-ui/04-components/03-message-list.md]: I think in the code sample there are nice comments which explain

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (ogkuzmin) [reply docusaurus/docs/Android/03-ui/04-components/03-message-list.md:121]: Are u sure that it works? My preview doesn't show it correctly

**Review by adasiewiczr — APPROVED** [review]
LGTM but let's wait for a second review from @nikomancy or @zsmb13

File: docusaurus/docs/Android/03-ui/04-components/03-message-list.md
Comment (zsmb13) [inline docusaurus/docs/Android/03-ui/04-components/03-message-list.md:53]: We should showcase using the factory instead of creating a ViewModel directly, otherwise clients are gonna copy this code and not have their ViewModels retained. We can probably also drop the Fragment context of this code snippet (especially since we're not initializing the `lateinit var` anyway) - see what the Channel List docs do for both of these.

**Review by zsmb13 — COMMENTED** [review]
The structure, the screenshots, the content is all great here! I pushed some edits to improve wording and formatting. Also renamed some headers to make them make more sense to users (what's a "message list item predicate" anyway).

I left one comment that I'll leave for you to check on @ogkuzmin , otherwise this is good to go 👍

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**
